tasks/export_abi.js

```diff
@@ -15,15 +16,17 @@ task(
     await hre.run(TASK_COMPILE, { noSizeContracts: true });
   }
 
-  const config = hre.config.abiExporter;
+  const configs = hre.config.abiExporter;
 
-  await hre.run('export-abi-group', { abiGroupConfig: config });
+  await Promise.all(configs.map(abiGroupConfig => {
+    return hre.run('export-abi-group', { abiGroupConfig })
+  }));
 });
 
 subtask(
   'export-abi-group'
 ).addParam(
-  'abiGroupConfig', undefined, undefined, Object
+  'abiGroupConfig', 'a single abi-exporter config object', undefined, types.any
```

Comment:
Can we use the `AbiExporterUserConfig` interface here instead of `types.any`?

Reply:
Related question, should we just convert the whole repository to ts?

Reply:
Unfortunately, hardhat defines these `types` weird. For example, this is `string`:

```ts
export const string: CLIArgumentType<string> = {
  name: "string",
  parse: (argName, strValue) => strValue,
  /**
   * Check if argument value is of type "string"
   *
   * @param argName {string} argument's name - used for context in case of error.
   * @param value {any} argument's value to validate.
   *
   * @throws HH301 if value is not of type "string"
   */
  validate: (argName: string, value: any): void => {
    const isString = typeof value === "string";

    if (!isString) {
      throw new HardhatError(ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE, {
        value,
        name: argName,
        type: string.name,
      });
    }
  },
};
```

We could either write our own, or use typescript (but that doesn't necessarily help people that are writing in plain JS).

Reply:
I see.  Let's leave it as-is.